src/Microsoft.ML.Tokenizers/PreTokenizer/Roberta.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Tokenizers
+{
+    /// <summary>
+    /// The pre-tokenizer for Roberta English tokenizer.
+    /// </summary>
+    public sealed class RobertaPreTokenizer : PreTokenizer
+    {
+        /// <summary>
+        /// Gets a singleton instance of the Roberta pre-tokenizer..
+        /// </summary>
+        public static readonly RobertaPreTokenizer Instance = new RobertaPreTokenizer();
+
+        private const string Pattern = @"'s|'t|'re|'ve|'m|'ll|'d| ?\p{L}+| ?\p{N}+| ?[^\s\p{L}\p{N}]+|\s+(?!\S)|\s+";
+
+        private static readonly IReadOnlyList<Split> _emptyList = new List<Split>();
+
+        /// <summary>
+        /// Splits the given string in multiple substrings at teh word boundary, keeping track of the offsets of said substrings from the original string.
```

Comment:
nit: `at the word boundary`

Reply:
This is also in the other "PreTokenizer" classes.

Reply:
good catch :-)